code/main/task/eventProcessor.brs

```diff
@@ -31,22 +31,57 @@ function _adb_EventProcessor(task as object) as object
 
             if _adb_isRequestEvent(event)
                 _adb_logInfo("handleEvent() - handle event: " + FormatJson(event))
-                if event.apiName = m._CONSTANTS.PUBLIC_API.SEND_EDGE_EVENT
-                    m._sendEvent(event)
-                    return
-                else if event.apiName = m._CONSTANTS.PUBLIC_API.SET_CONFIGURATION
-                    m._setConfiguration(event)
-                    return
-                else if event.apiName = m._CONSTANTS.PUBLIC_API.SET_LOG_LEVEL
-                    m._setLogLevel(event)
-                    return
-                else if event.apiName = m._CONSTANTS.PUBLIC_API.SET_EXPERIENCE_CLOUD_ID
-                    m._setECID(event)
-                    return
-                else if event.apiName = m._CONSTANTS.PUBLIC_API.RESET_IDENTITIES
-                    m._resetIdentities(event)
-                    return
+                if m._task <> invalid and m._task.hasField("debugInfo")
+                    loggingService = _adb_serviceProvider().loggingService
+
+                    networkService = _adb_serviceProvider().networkService
+                    networkService._debugMode = true
+                    ' -------------------
+                    m._processAdbRequestEvent(event)
+                    ' -------------------
+                    m._dumpDebugInfo(event, loggingService, networkService)
+                else
+                    m._processAdbRequestEvent(event)
                 end if
+
+            else
+                _adb_logWarning("handleEvent() - event is invalid: " + FormatJson(event))
+            end if
+        end function,
+
+        _dumpDebugInfo: function(event as object, loggingService as object, networkService as object) as void
+            debugInfo = {
+                eventId: event.uuid,
+                apiName: event.apiName
+            }
+
+            debugInfo.logLevel = loggingService.getLogLevel()
+            if _adb_isConfigurationModule(m._configurationModule)
```

Comment:
Where is this` m._configurationModule` set? We can by default add all the dumps without these checks right?

Reply:
We discussed the option of enabling selected edge modules, so we need to perform invalid checks here.

Reply:
We can dump everything as we don't have a usecase of delivering only selected modules. And by default we will have all the modules at all the times.